Document/0x05f-Testing-Local-Authentication.md

```diff
@@ -82,21 +84,66 @@ Make sure that the unlocked key is used during the application flow. For example
 
 #### Dynamic Analysis
 
-Patch the app or use runtime instrumentation to bypass fingerprint authentication on the client. For example, you could use Frida to call the `onActivityResult` callback method directly to see if the cryptographic material (e.g. the setup cipher) can be ignored to proceed with the local authentication flow. Refer to the chapter "[Tampering and Reverse Engineering on Android](0x05c-Reverse-Engineering-and-Tampering.md)" for more information.
+Validate the duration of time (seconds) for which the key is authorized to be used after the user is successfully authenticated. This is only needed if `setUserAuthenticationRequired` is used.
 
 ### Testing Biometric Authentication (MSTG-AUTH-8)
 
 #### Overview
 
-Android 6.0 (API level 23) introduced public APIs for authenticating users via fingerprint. Access to the fingerprint hardware is provided through the [FingerprintManager class](https://developer.android.com/reference/android/hardware/fingerprint/ "FingerprintManager"). An app can request fingerprint authentication by instantiating a `FingerprintManager` object and calling its `authenticate` method. The caller registers callback methods to handle possible outcomes of the authentication process (i.e. success, failure, or error). Note that this method doesn't constitute strong proof that fingerprint authentication has actually been performed - for example, the authentication step could be patched out by an attacker, or the "success" callback could be called using instrumentation.
+Biometric authentication allows a convenient mechanism for authentication, but also introduces additional attack surface when using it. The Android developer documentation gives an interesting overview and indicators for [measuring biometric unlock security](https://source.android.com/security/biometric/measure#strong-weak-unlocks "Measuring Biometric Unlock Security").
+
+On the Android platform there are three different classes available for biometric authentication:
+
+- Android Q / 10 (API level 29) and higher: `BiometricManager`

```

Comment:
SG and as in 0x5a:
```suggestion
- Android 10 (API level 29) and higher: `BiometricManager`
```

Reply:
Yes, you are right, but I put this in there on purpose, as the Screenshot above is using "Android Q", so it's easier to reference to this screenshot. Otherwise you are lost in the Android version jungle. Therefore I would keep it like it is and make an exception. Ok for you?

Reply:
Aha, I see. Yes then I'm also fine with it :)